torch/_prims_common/wrappers.py

```diff
@@ -298,7 +298,19 @@ def _fn(*args, out=None, **kwargs):
             annotation=out_type,
         )
         # Mark that the function now returns a tuple
-        assert sig.return_annotation in (sig.empty, out_type)
+        if (
+            isinstance(sig.return_annotation, str)
+            and "torch._tensor.Tensor" in sig.return_annotation
+        ):
+            # FIXME(T168485992): This is a forward reference, i.e a type
+            # annotation wrapped in a string literal. In theory, we would need
+            # to parse it, but we don't have the global context (i.e imports)
+            # to do that correctly. Since we know it is most likely introduced
+            # by `pyre infer` before running Pysa, so we can safely ignore it
+            # for now.
+            pass
+        else:
+            assert sig.return_annotation in (sig.empty, out_type)
```

Comment:
Annotations can be strings in some cases, which is fair, so what about simply
```suggestion
        assert isinstance(sig.return_annotation, str) or sig.return_annotation in (sig.empty, out_type)
```
This is a bit more forward-looking as it also covers functions thay may not return tensors.

Reply:
For reference, this diff was unfortunately landed internally a couple days ago and there's a followup diff which adds annotations D50847843

Reply:
I'm not completely sure what these things imply